### PR TITLE
Numpy compatible eye

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
Numpy compatible eye

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
@haojin2 @reminisce

### CODE REVIEW

File: src/operator/numpy/np_init_op.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  CHECK(param.N >= 0) << "negative dimensions are not allowed. N is " << param.N;
+  CHECK(M >= 0) << "negative dimensions are not allowed. M is " << M;
+  SHAPE_ASSIGN_CHECK(*out_attrs, 0, mshadow::Shape2(param.N, M));
+
+  return out_attrs->at(0).ndim() != 0U;
+}
+
+template<typename xpu>
+void NumpyEyeFill(const nnvm::NodeAttrs& attrs,
+             const OpContext& ctx,
```
Comment (haojin2) [inline src/operator/numpy/np_init_op.h]: nit: align all arguments with the line above like:
```c++
void NumpyEyeFill(const nnvm::NodeAttrs& attrs,
                  const OpContext& ctx,
```
for every function here.

Comment (stu1130) [reply src/operator/numpy/np_init_op.h]: done

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -690,3 +690,34 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
...
+        and a negative value to a lower diagonal.
+    dtype : data-type, optional
+        Data-type of the returned array.
+
+    Returns
+    -------
+    I : ndarray of shape (N,M)
+        An array where all elements are equal to zero,
+        except for the k-th diagonal, whose values are equal to one.
+    """
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: nit: one more blank line below.

**Comment by haojin2** [comment]
@stu1130 Please rebase and resolve conflict when you have time, thx

**Pull request merged**